components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -1513,82 +1512,39 @@ func socketActivationForDocker(ctx context.Context, wg *sync.WaitGroup, term *te
 	}
 }
 
-func analyseConfigChanges(ctx context.Context, wscfg *Config, w analytics.Writer, cfgobs config.ConfigInterface) {
-	cfgc := cfgobs.Observe(ctx)
-	var (
-		cfg     *gitpod.GitpodConfig
-		t       = time.NewTicker(10 * time.Second)
-		changes []string
-	)
-	defer t.Stop()
-
-	computeHash := func(i interface{}) (string, error) {
-		b, err := json.Marshal(i)
-		if err != nil {
-			return "", err
-		}
-		h := sha256.New()
-		_, err = h.Write(b)
-		if err != nil {
-			return "", err
-		}
-		return fmt.Sprintf("%x", h.Sum(nil)), nil
+func analyseConfigChanges(ctx context.Context, wscfg *Config, w analytics.Writer, cfgobs config.ConfigInterface, gitpodAPI gitpod.APIInterface) {
+	info, err := gitpodAPI.GetWorkspace(ctx, wscfg.WorkspaceID)
+	if err != nil {
+		log.WithError(err).Error("gitpod config analytics: failed to track config changes")
+		return
 	}
 
+	var analyzer *config.ConfigAnalyzer
+	log.Debug("gitpod config analytics: watching...")
+
+	cfgs := cfgobs.Observe(ctx)
 	for {
 		select {
-		case c, ok := <-cfgc:
+		case cfg, ok := <-cfgs:
 			if !ok {
 				return
 			}
-			if cfg == nil {
-				cfg = c
-				continue
+			if analyzer != nil {
+				analyzer.Analyse(cfg)
+			} else {
+				analyzer = config.NewConfigAnalyzer(log.Log, 5*time.Second, func(field string) {
+					w.Track(analytics.TrackMessage{
+						Identity: analytics.Identity{UserID: info.Workspace.OwnerID},
+						Event:    "gitpod_config_changed",
+						Properties: map[string]interface{}{
+							"key":         field,
```

Comment:
**Observation (non-blocking)**: Opening a workspace without .gitpod.yml and then running `gp init` triggered 6 `gitpod_config_changed` at once, one for each of the keys:
- MainConfiguration
- WorkspaceLocation
- Ports
- CheckoutLocation
- Tasks
- ExperimentalNetwork

<img width="507" alt="image" src="https://user-images.githubusercontent.com/418083/183590308-b86f4f80-8cf8-4875-a690-43b6e21f1f26.png">

The content of the generated .gitpod.yml was the following:

```yml
# List the start up tasks. Learn more https://www.gitpod.io/docs/config-start-tasks/
tasks:
  - init: echo 'init script' # runs during prebuild
    command: echo 'start script'

# List the ports to expose. Learn more https://www.gitpod.io/docs/config-ports/
ports:
  - port: 3000
    onOpen: open-preview
```

So I expected it would trigger just two events keys: "Tasks" and "Ports".

Reply:
rrr, it is a bug, let me have a look a bit later, what is going on